evm/src/NttManager/NttManagerWithPerChainTransceivers.sol

```diff
@@ -0,0 +1,165 @@
+        Mode _mode,
+        uint16 _chainId
+    ) NttManagerNoRateLimiting(_token, _mode, _chainId) {}
+
+    // ==================== Storage slots ==========================================================
+
+    bytes32 private constant SEND_TRANSCEIVER_BITMAP_SLOT =
+        bytes32(uint256(keccak256("nttpct.sendTransceiverBitmap")) - 1);
+
+    bytes32 private constant RECV_TRANSCEIVER_BITMAP_SLOT =
+        bytes32(uint256(keccak256("nttpct.recvTransceiverBitmap")) - 1);
+
+    // ==================== Override / implementation of transceiver stuff =========================
+
+    /// @inheritdoc IManagerBase
+    function enableSendTransceiverForChain(
+        address transceiver,
+        uint16 forChainId
+    ) external override(ManagerBase, IManagerBase) onlyOwner {
+        _enableTransceiverForChain(transceiver, forChainId, SEND_TRANSCEIVER_BITMAP_SLOT);
+        emit SendTransceiverEnabledForChain(transceiver, forChainId);
+    }
+
+    /// @inheritdoc IManagerBase
+    function enableRecvTransceiverForChain(
+        address transceiver,
+        uint16 forChainId
+    ) external override(ManagerBase, IManagerBase) onlyOwner {
+        _enableTransceiverForChain(transceiver, forChainId, RECV_TRANSCEIVER_BITMAP_SLOT);
+        emit RecvTransceiverEnabledForChain(transceiver, forChainId);
+    }
+
+    function _enableTransceiverForChain(
+        address transceiver,
+        uint16 forChainId,
+        bytes32 tag
+    ) private {
+        if (transceiver == address(0)) {
+            revert InvalidTransceiverZeroAddress();
+        }
+
+        mapping(address => TransceiverInfo) storage transceiverInfos = _getTransceiverInfosStorage();
+        if (!transceiverInfos[transceiver].registered) {
+            revert NonRegisteredTransceiver(transceiver);
+        }
+
+        uint8 index = _getTransceiverInfosStorage()[transceiver].index;
+        mapping(uint16 => _EnabledTransceiverBitmap) storage _bitmaps =
+            _getPerChainTransceiverBitmapStorage(tag);
+        _bitmaps[forChainId].bitmap |= uint64(1 << index);
+    }
+
+    function _isSendTransceiverEnabledForChain(
+        address transceiver,
+        uint16 forChainId
+    ) internal view override returns (bool) {
+        uint64 bitmap = _getPerChainTransceiverBitmap(forChainId, SEND_TRANSCEIVER_BITMAP_SLOT);
+        uint8 index = _getTransceiverInfosStorage()[transceiver].index;
+        return (bitmap & uint64(1 << index)) != 0;
+    }
+
+    function _getEnabledRecvTransceiversForChain(
+        uint16 forChainId
+    ) internal view returns (uint64 bitmap) {
+        return _getPerChainTransceiverBitmap(forChainId, RECV_TRANSCEIVER_BITMAP_SLOT);
+    }
+
+    function _getPerChainTransceiverBitmap(
+        uint16 forChainId,
+        bytes32 tag
+    ) private view returns (uint64 bitmap) {
+        bitmap = _getPerChainTransceiverBitmapStorage(tag)[forChainId].bitmap;
+        if (bitmap == 0) {
+            bitmap = type(uint64).max;
+        }
```

Comment:
Doesn't this default to all registered transceivers being enabled, even if they were disabled? Should this instead load from `_getTransceiverBitmapStorage`?

Reply:
As discussed below, this saves gas by avoid an extra storage slot read, and it works because the value gets combined with the bitmap of enabled transceivers. I added a comment to reflect that.